Tangem/Modules/Markets/Providers/MarketsListDataProvider.swift

```diff
@@ -19,23 +19,42 @@ final class MarketsListDataProvider {
     // MARK: Published Properties
 
     @Published var items: [MarketsTokenModel] = []
+    @Published var isLoading: Bool = false
 
     // MARK: - Public Properties
 
     var lastSearchTextValue: String? {
         return lastSearchText
     }
 
-    // Tells if all items have been loaded. (Used to hide/show activity spinner)
-    private(set) var canFetchMore = true
+    var lastFilterValue: Filter? {
+        return lastFilter
+    }
+
+    // Tells if all items have been loaded
+    var canFetchMore: Bool {
+        guard !isLoading else {
```

Comment:
думаю тут нормально подходит:
```suggestion
        if isLoading {
```

а не будет проблемы, что загрузчик будет пропадать на момент загрузки новой страницы?